kotlinx-coroutines-test/common/src/TestScope.kt

```diff
@@ -186,11 +186,20 @@ internal class TestScopeImpl(context: CoroutineContext) :
             finished = true
             uncaughtExceptions
         }
-        if (exceptions.isEmpty() && (children.any { it.isActive } || !testScheduler.isIdle()))
-            throw UncompletedCoroutinesError(
-                "Unfinished coroutines during teardown. Ensure all coroutines are" +
-                    " completed or cancelled by your test."
-            )
+        val activeJobs = children.filter { it.isActive }.toList()
```

Comment:
It seems like `it.isCompleted` should be used instead, as `isActive` being `false` may imply that the coroutine is in its cancellation sequence and still is executing.

Though I can't see how it is possible to trigger this check at all, could you please ensure it's still needed?

Reply:
Probably it can be triggered by the timeout, but I neither see any tests nor sure that this exception will be actually the final one

Reply:
If a timeout happened, this wouldn't be the final exception, no, it throws a more informative message instead.

I added this for use with `runBlockingTest`, which now supports using `TestScope` but doesn't properly await completion. Though looking at it, I think that it would just throw `IllegalStateException` in this case instead. Fixed it and added comments.